src/main/java/org/dataone/portal/TokenGenerator.java

```diff
@@ -181,35 +184,52 @@ public String getJWT(String userId, String fullName)
     private void setPrivateKey() throws IOException {
         String privateKeyFileName =
             Settings.getConfiguration().getString("cn.server.privatekey.filename");
-        String privateKeyPassword = null;
 
         CertificateManager cmInst = CertificateManager.getInstance();
         // consumers do not need the private key
         if (privateKeyFileName != null) {
-            privateKey = (RSAPrivateKey) cmInst.loadPrivateKeyFromFile(privateKeyFileName,
-                                                                       privateKeyPassword);
+            // uses a null password:
+            privateKey = (RSAPrivateKey) cmInst.loadPrivateKeyFromFile(privateKeyFileName, null);
         }
     }
 
     /*
      * Set the public key
      * @throws IOException
      */
-    private void setPublicKey() throws IOException {
-        // use either the configured certificate, or fetch it from the CN
-        String certificateFileName =
-            Settings.getConfiguration().getString("cn.server.publiccert.filename");
+    protected void setPublicKeys() throws IOException {
```

Comment:
Do we need to think about the public key order in the list? The most used one ( which maps the private key to sign token) should be put in the first one. So the verify process can be more efficient.

Reply:
done 2d638c7a2c57ae4e9ef92c6b114c33716807ee53